code/modules/client/client_procs.dm

```diff
@@ -912,14 +926,21 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 		var/second = round(world.time, 10)
 		if (!clicklimiter)
 			clicklimiter = new(LIMITER_SIZE)
+
 		if (second != clicklimiter[CURRENT_SECOND])
 			clicklimiter[CURRENT_SECOND] = second
 			clicklimiter[SECOND_COUNT] = 0
-		clicklimiter[SECOND_COUNT] += 1+(!!ab)
+
+		clicklimiter[SECOND_COUNT] += 1 + (!!ab)
+
 		if (clicklimiter[SECOND_COUNT] > scl)
 			to_chat(src, span_danger("Your previous click was ignored because you've done too many in a second"))
 			return
 
+	//check if the server is overloaded and if it is then queue up the click for next tick
+	if(SSinput.queue_click_from_usr(object, location, control, params))
+		return
```

Comment:
Clicks are likelyish to read to icon2htmls, which can be quite expensive. I guess this check will never pass in high maptick cases, but what about right below that line?

Reply:
if its below the threshold then it continues the click as normal